libr/anal/var.c

```diff
@@ -685,26 +685,53 @@ R_API void extract_rarg(RAnal *anal, RAnalOp *op, RAnalFunction *fcn, int *reg_s
 	if (op->dst) {
 		opdreg = get_regname (anal, op->dst);
 	}
-	if (opsreg) {
-		RRegItem *ri = r_reg_get (anal->reg, opsreg, -1);
-		if (ri) {
-			delta = ri->index;
-		}
-	}
 	for (i = 0; i < max_count; i++) {
 		const char *regname = r_anal_cc_arg (anal, fcn->cc, i + 1);
 		// reg_set enusres we only extract first-read argument reg
 		if (!reg_set [i] && regname) {
-			if (op->src[0] && opsreg && !strcmp (opsreg, regname)) {
-				char *vname = r_str_newf ("%s%d", "arg", i + 1);
-				r_anal_var_add (anal, fcn->addr, 1, delta, 'r', NULL, anal->bits / 8,
+			bool cond = (op->type  == R_ANAL_OP_TYPE_CMP) && opdreg &&
+				!strcmp (opdreg, regname);
+			if ((op->src[0] && opsreg && !strcmp (opsreg, regname)) || cond) {
+				const char *vname;
+				const char *type = "int";
+				char *fname = fcn->name;
+				char *name = NULL;
+				int delta;
+				RRegItem *ri = r_reg_get (anal->reg, regname, -1);
+				if (ri) {
+					delta = ri->index;
+				}
+				if (fname) {
+					char *tmp = strchr (fname, '.');
+					if (tmp) {
+						fname = tmp + 1;
+					}
+				}
+				if ((i < 3) && fname && !strcmp(fname, "main")) {
```

Comment:
Why hardcoding it here and not using the types database?

Reply:
afaik , we don't have `main` function info in types sdb , so hardcoded it !

Reply:
Then what about adding it then there instead of hardcoding!? 🤔

![image](https://user-images.githubusercontent.com/1408600/43831426-60caa60e-9b04-11e8-86a5-14f03ee65a45.png)

Reply:
:D , ok !!